### PR TITLE
Win and macOS: Wrong language selected when region not default

### PR DESCRIPTION
**Pull request by danyeaw**
@mathiascode pointed out (thanks!) in https://github.com/gaphor/gaphor/pull/1973 that `locale.getlocale()` returns the wrong language if the region in Windows or macOS is set different from the display language.

This PR uses the native language defaults in Windows and macOS to get the correct language.

I feel like this should just be built in to Python, maybe we should attempt to create a PR to CPython itself?

### PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I have read, and I am following the [Contributor guide](https://github.com/gaphor/gaphor/blob/main/CONTRIBUTING.md)
- [X] I have read, and I understand the [Code of Conduct](https://github.com/gaphor/gaphor/blob/main/CODE_OF_CONDUCT.md)

### PR Type
What kind of change does this PR introduce?

- [X] Bug fix
- [ ] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?
The wrong language is selected under certain circumstances.

Issue Number: N/A

### What is the new behavior?
The correct language should always be selected.

### Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

File: gaphor/i18n.py [hunk 1]
```diff
@@ -15,18 +17,47 @@
...
+            return language
+    elif sys.platform == "darwin":
+        import subprocess
+
+        try:
+            lang_out = subprocess.check_output(
+                ("defaults", "read", "-g", "AppleLanguages")
+            )
+            if languages := lang_out.decode("utf-8").strip('()\n" ').split(","):
+                return languages[0][:2]
```
Comment (mathiascode) [inline gaphor/i18n.py:42]: This output works with the LANGUAGE environment variable, but I don't remember if LANG supports it. Might be worth confirming this.

**Comment by amolenaar** [comment]
What settings for region (=country?) and language should be set differently to make locale.getlocale() produce the wrong locale?

File: gaphor/i18n.py (hunk 1 above)
Comment (danyeaw) [reply gaphor/i18n.py:42]: I think LANGUAGE allows for a list of languages, but I expect here we are just getting the first one. Maybe @amolenaar would be able to help test on macOS that this works.

**Comment by mathiascode** [comment]
> I feel like this should just be built in to Python, maybe we should attempt to create a PR to CPython itself?

I'm not sure if there are other cross-platform Python projects that handle this better, but it would be nice to know how others do translations. The macOS hack in particular is very ugly, but at the time I didn't find another way without bringing in third-party dependencies.

**Comment by danyeaw** [comment]
> What settings for region (=country?) and language should be set differently to make locale.getlocale() produce the wrong locale?

In Windows, if I set the language to Spanish and the region to Ingles with any country, Gaphor starts in English.

File: gaphor/i18n.py (hunk 1 above)
Comment (amolenaar) [reply gaphor/i18n.py:42]: Maybe https://stackoverflow.com/questions/17679651/how-to-figure-out-what-the-current-system-preferred-language-is? It isn't spawning any processes at least.

Comment (danyeaw) [reply gaphor/i18n.py:42]: Nice one, I'll update the PR!

**Comment by amolenaar** [comment]
I can confirm that on macOS, I always get the message on the `main` branch

```
No translations were found: [Errno 2] No translation file found for domain: 'gaphor'
```

With the changes on this PR it's taking my preferred language, which I set to Dutch for this test. I set my region (country) to Spain. The language I see is Dutch, as expected.

I added a commit, so only the Cocoa framework code is installed.

**Comment by amolenaar** [comment]
I think it's good to merge.

**Comment by danyeaw** [comment]
Thanks for the teamwork on this one!

**Pull request merged**